src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMappingAdjournmentTest.java

```diff
@@ -0,0 +1,185 @@
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import uk.gov.hmcts.reform.sscs.ccd.domain.AdjournCaseNextHearingDurationType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.AdjournCaseNextHearingDurationUnits;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Appeal;
+import uk.gov.hmcts.reform.sscs.ccd.domain.BenefitCode;
+import uk.gov.hmcts.reform.sscs.ccd.domain.HearingOptions;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Issue;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.domain.YesNo;
+import uk.gov.hmcts.reform.sscs.reference.data.model.HearingDuration;
+import uk.gov.hmcts.reform.sscs.service.holder.ReferenceDataServiceHolder;
+
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.BDDMockito.given;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.AdjournCaseNextHearingDurationUnits.MINUTES;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.AdjournCaseNextHearingDurationUnits.SESSIONS;
+
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class HearingsDurationMappingAdjournmentTest extends HearingsMappingBase {
+
+    @Mock
+    private ReferenceDataServiceHolder referenceDataServiceHolder;
+
+    private void setAdjournmentDurationAndUnits(Integer duration, AdjournCaseNextHearingDurationUnits units) {
+        caseData.getAdjournment().setNextHearingListingDuration(duration);
+        caseData.getAdjournment().setNextHearingListingDurationUnits(units);
+    }
+
+    @BeforeEach
+    void setUp() {
+        given(referenceDataServiceHolder.isAdjournmentFlagEnabled()).willReturn(true);
+
+        caseData.getAdjournment().setNextHearingListingDurationType(AdjournCaseNextHearingDurationType.NON_STANDARD);
+        caseData.getAdjournment().setAdjournmentInProgress(YesNo.YES);
+    }
+
+    @DisplayName("When a valid adjournCaseDuration and adjournCaseDurationUnits is given "
+        + "getHearingDuration returns the correct duration Parameterized Tests")
+    @ParameterizedTest
+    @MethodSource
+    void getHearingDuration(
+        Integer adjournCaseDuration,
+        AdjournCaseNextHearingDurationUnits adjournCaseDurationUnits,
+        int expected
+    ) {
+        setAdjournmentDurationAndUnits(adjournCaseDuration, adjournCaseDurationUnits);
+
+        Integer result = HearingsDurationMapping.getHearingDuration(caseData, referenceDataServiceHolder);
+
+        assertThat(result).isEqualTo(expected);
+    }
+
+    private Stream<Arguments> getHearingDuration() {
+        return Stream.of(
+            Arguments.of(120, MINUTES, 120),
+            Arguments.of(70, MINUTES, 70),
+            Arguments.of(1, SESSIONS, 165),
+            Arguments.of(2, SESSIONS, 330),
+            Arguments.of(3, SESSIONS, 495)
+        );
+    }
```

Comment:
Really not a fan of arguments being handled like this, can you use csv source or split the test instead?

Reply:
Done